scipy/integrate/tests/test_quadrature.py

```diff
@@ -180,6 +179,18 @@ def test_simpson(self):
         assert_equal(simpson(y, x=x, axis=0), zero_axis)
         assert_equal(simpson(y, x=x, axis=-1), default_axis)
 
+    @pytest.mark.parametrize('droplast', [False, True])
+    def test_simpson_2d_integer_no_x(self, droplast):
+        # The inputs are 2d integer arrays.  The results should be
+        # identical to the results when the inputs are floating point.
+        y = np.array([[2, 2, 4, 4, 8, 8, -4, 5],
+                      [4, 4, 2, -4, 10, 22, -2, 10]])
+        if droplast:
+            y = y[:, :-1]
+        result = simpson(y, axis=-1)
+        expected = simpson(np.array(y, dtype=np.float64), axis=-1)
+        assert_equal(result, expected)
```

Comment:
The doc change above has `simpson` returning a `float` -- should we use `assert_allclose()` here instead then?

Apart from that, this PR just adds a new test and changes `4` to `4.0` in the source, so it is hard to imagine a drawback, even as a non-expert.

Reply:
I think `assert_equal` is fine here.  The integer input will end being cast to float internally, so in this case, it should end up doing exactly the same calculation on exactly the same floating point arrays.